research/gnn-survey/models.py

```diff
@@ -89,3 +96,78 @@ def call(self, inputs):
     x = (features, adj)
     outputs = self.classifier(x)
     return outputs
+
+class GATBlock(tf.keras.layers.Layer):
+  """Graph attention block."""
+
+  def __init__(self, hidden_dim, dropout_rate, num_heads, **kwargs):
+    """Initializes a GGN block.
+
+    Args:
+      hidden_dim: (int) Dimension of hidden layer.
+      dropout_rate: (float) Dropout probability
+      **kwargs: Keyword arguments for tf.keras.layers.Layer.
+    """
+    super(GATBlock, self).__init__(**kwargs)
+    self.hidden_dim = hidden_dim
+    self.dropout_rate = dropout_rate
+    self.num_heads = num_heads
+
+    self._activation = tf.keras.layers.ELU()
+    self._dropout = tf.keras.layers.Dropout(self.dropout_rate)
+
+  def build(self, input_shape):
+    super(GATBlock, self).build(input_shape)
+    self._graph_attn_layer = [
```

Comment:
If it doesn't depend on input shape, do this in __init__

Reply:
OK.